doc/common_pitfalls.rst

```diff
@@ -82,34 +91,31 @@ on actually novel data, for example during production.
 
 A common cause is not keeping the test and train data subsets separate.
 Test data should never be used to make choices about the model.
-The general rule is to never call `fit` on test data. This is particularly
-important for 'supervised transformations' where the `fit` method requires
-both the data, `X` and the target values, `y`. While this may
-sound obvious, this is easy to miss in some cases, for example when applying
-certain pre-processing steps.
-
-Although both train and test data subsets should receive the same preprocessing
-transformation, it is important that these transformations are only learnt
-from the training data. For example, if you have a
+**The general rule is to never call** `fit` **on the test data**. While this
+may sound obvious, this is easy to miss in some cases, for example when
+applying certain pre-processing steps.
+
+Although both train and test data subsets should receive the same
+preprocessing transformation (as described in the previous section), it is
+important that these transformations are only learnt from the training data.
+For example, if you have a
 normalization step where you divide by the average value, the average should
 be the average of the train subset, **not** the average of all the data. If the
 test subset is included in the average calculation, information from the test
 subset is influencing the model.
 
-Including the test data when :ref:`tuning model hyperparameters <grid_search>`
-will also inadvertently introduce information from the test data into the
-model. Practically, this means that only the train data subset should be fed
-into the `fit` method of :ref:`hyper_parameter_optimizers`, as is the case
-with all scikit-learn estimators.
-
 An example of data leakage during preprocessing is detailed below.
 
-.. topic:: See Also:
+Data leakage during pre-processing
+----------------------------------
 
-  * :ref:`sphx_glr_auto_examples_model_selection_plot_nested_cross_validation_iris.py`
```

Comment:
This example isn't about leaking the testing data, but rather about the distinction between model training, model selection, and model evaluation.

This is a related but distinct concept (which would make a great section here BTW).

Reply:
But do you think the example is adequate and if we added a section it would be repeating ourselves?

Reply:
In an ideal world I would make a new section here (ignoring the example), and then consider whether it's worth modifying or even removing the example.

Having a section here will help visibility and this is an important topic.

Also, the example doesn't do a great job IMHO at explaining what the core message is. It's talking about nested vs non nested CV, but in reality the goal was to illustrate how you should never do model fitting, model selection and model evaluation on the same datasets. This is at least what I understand from the original PR where the example comes from.

So I'd prioritize a new section here over the example.

Reply:
If we were to add a new section that would link to the nested CV example, I think that would be a section about over-fitting via automated parameter selection. However I believe it will be hard to demonstrate a significant effect in a fast running code snippet.